src/util/dom_misc.ts

```diff
@@ -1,301 +1,144 @@
-
-  /**
-   * Returns element scroll offsets
-   * @memberOf fabric.util
-   * @param {HTMLElement} element Element to operate on
-   * @return {Object} Object with left/top values
-   */
-  function getScrollLeftTop(element) {
-
-    var left = 0,
-        top = 0,
-        docElement = fabric.document.documentElement,
+  wrapper.appendChild(element);
+  return wrapper;
+}
+
+/**
+ * Returns element scroll offsets
+ * @memberOf fabric.util
+ * @param {HTMLElement} element Element to operate on
+ * @return {Object} Object with left/top values
+ */
+export function getScrollLeftTop(element) {
+
+  let left = 0,
+      top = 0;
+
+  const docElement = fabric.document.documentElement,
         body = fabric.document.body || {
           scrollLeft: 0, scrollTop: 0
         };
-
-    // While loop checks (and then sets element to) .parentNode OR .host
-    //  to account for ShadowDOM. We still want to traverse up out of ShadowDOM,
-    //  but the .parentNode of a root ShadowDOM node will always be null, instead
-    //  it should be accessed through .host. See http://stackoverflow.com/a/24765528/4383938
-    while (element && (element.parentNode || element.host)) {
-
-      // Set element to element parent, or 'host' in case of ShadowDOM
-      element = element.parentNode || element.host;
-
-      if (element === fabric.document) {
-        left = body.scrollLeft || docElement.scrollLeft || 0;
-        top = body.scrollTop ||  docElement.scrollTop || 0;
-      }
-      else {
-        left += element.scrollLeft || 0;
-        top += element.scrollTop || 0;
-      }
-
-      if (element.nodeType === 1 && element.style.position === 'fixed') {
-        break;
-      }
+  // While loop checks (and then sets element to) .parentNode OR .host
+  //  to account for ShadowDOM. We still want to traverse up out of ShadowDOM,
+  //  but the .parentNode of a root ShadowDOM node will always be null, instead
+  //  it should be accessed through .host. See http://stackoverflow.com/a/24765528/4383938
+  while (element && (element.parentNode || element.host)) {
+
+    // Set element to element parent, or 'host' in case of ShadowDOM
+    element = element.parentNode || element.host;
+
+    if (element === fabric.document) {
+      left = body.scrollLeft || docElement.scrollLeft || 0;
+      top = body.scrollTop ||  docElement.scrollTop || 0;
+    }
+    else {
+      left += element.scrollLeft || 0;
+      top += element.scrollTop || 0;
     }
 
-    return { left: left, top: top };
+    if (element.nodeType === 1 && element.style.position === 'fixed') {
+      break;
+    }
   }
 
-  /**
-   * Returns offset for a given element
-   * @function
-   * @memberOf fabric.util
-   * @param {HTMLElement} element Element to get offset for
-   * @return {Object} Object with "left" and "top" properties
-   */
-  function getElementOffset(element) {
-    var docElem,
-        doc = element && element.ownerDocument,
-        box = { left: 0, top: 0 },
+  return { left, top };
```

Comment:
can this be a point as well?

Reply:
not sure, this is style related, it does not need to be transformed or anything else.